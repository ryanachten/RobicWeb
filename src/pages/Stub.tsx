import * as React from "react";
import { Theme } from "@material-ui/core/styles/createMuiTheme";
import createStyles from "@material-ui/core/styles/createStyles";
import withStyles from "@material-ui/core/styles/withStyles";
import Typography from "@material-ui/core/Typography";

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: "center",
      display: "flex",
      height: "100vh",
      justifyContent: "center",
      width: "100vw"
    }
  });

type State = {};

type Props = {
  classes: any;
  pageName: string;
};

class Stub extends React.Component<Props, State> {
  render() {
    const { classes, pageName } = this.props;
    return (
      <div className={classes.root}>
        <Typography variant="h1">{pageName}</Typography>
      </div>
    );
  }
}

export default withStyles(styles)(Stub);
