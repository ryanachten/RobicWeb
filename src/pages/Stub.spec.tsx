import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import Stub from "./Stub";
import renderer from "react-test-renderer";

describe("Page: Stub", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow({ untilSelector: "Stub" });
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Stub pageName="Stub" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
