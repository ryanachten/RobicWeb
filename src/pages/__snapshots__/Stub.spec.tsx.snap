// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Stub renders correctly 1`] = `
<div
  className="Stub-root-1"
>
  <h1
    className="MuiTypography-root-2 MuiTypography-h1-14"
  >
    Stub
  </h1>
</div>
`;
